lib/chef/provider/aws_vpc.rb

```diff
@@ -148,47 +149,36 @@ def update_vpc_attributes(vpc)
   def update_internet_gateway(vpc)
     current_ig = vpc.internet_gateway
     case new_resource.internet_gateway
-    when String, Chef::Resource::AwsInternetGateway, AWS::EC2::InternetGateway
-      new_ig = Chef::Resource::AwsInternetGateway.get_aws_object(new_resource.internet_gateway, resource: new_resource)
-      if !current_ig
-        converge_by "attach Internet gateway #{new_resource.internet_gateway} to VPC #{vpc.id}" do
-          new_ig.attach(vpc.id)
-        end
-      elsif current_ig != new_ig
-        converge_by "replace Internet gateway #{current_ig.id} on VPC #{vpc.id} with new Internet gateway #{new_ig.id}" do
-          current_ig.detach(vpc.id)
-          new_ig.attach(vpc.id)
-        end
-        if current_ig.tags['OwnedByVPC'] == vpc.id
-          converge_by "destroy Internet gateway #{current_ig.id} in #{region} (owned by VPC #{vpc.id})" do
-            current_ig.delete
+      when String, Chef::Resource::AwsInternetGateway, AWS::EC2::InternetGateway
+        new_ig = Chef::Resource::AwsInternetGateway.get_aws_object(new_resource.internet_gateway, resource: new_resource)
+        if !current_ig
+          aws_internet_gateway new_ig do
+            vpc vpc.id
           end
-        end
-      end
-    when true
-      if !current_ig
-        converge_by "attach Internet gateway to VPC #{vpc.id}" do
-          current_ig = AWS.ec2(config: vpc.config).internet_gateways.create
-          retry_with_backoff(NeverObtainedExistence) do
-            raise NeverObtainedExistence.new("Internet gateway for VPC #{vpc.id} never obtained existence") unless current_ig.exists?
+        elsif current_ig != new_ig
+          aws_internet_gateway current_ig do
+            if current_ig.tags['OwnedByVPC'] == vpc.id
+              action :destroy
+            else
+              action :detach
+            end
```

Comment:
It looks like thei logic here is missing attaching the `new_ig`.  I think you could make lines 154-166 look like

```ruby
if current_ig != new_ig
  if current_ig
    aws_internet_gateway current_ig do
      if current_ig.tags['OwnedByVPC'] == vpc.id
        action :destroy
      else
        action :detach
      end
    end
  end
  aws_internet_gateway new_ig do
    vpc vpc.id
  end
end
```